metricq_tools/summary.py

```diff
@@ -0,0 +1,339 @@
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from contextlib import suppress
+
+import asyncio
+import aio_pika
+import click
+import click_completion
+import click_log
+import metricq
+import numpy as np
+import termplotlib as tpl
+from metricq.datachunk_pb2 import DataChunk
+
+from metricq_tools.utils import metricq_server_option, metricq_token_option
+
+from .logging import get_root_logger
+from .version import version as client_version
+
+logger = get_root_logger()
+
+click_completion.init()
+
+
+class SummarySink(metricq.Sink):
+    def __init__(
+        self,
+        metrics: list[str],
+        intervals_histogram: bool,
+        chunk_sizes_histogram: bool,
+        values_histogram: bool,
+        print_stats: bool,
+        print_data: bool,
+        command: str,
+        *args,
+        **kwargs,
+    ):
+        self._metrics = metrics
+        self.tokens = {}
+
+        self.print_intervals = intervals_histogram
+        self.print_chunk_sizes = chunk_sizes_histogram
+        self.print_values = values_histogram
+        self.print_stats = print_stats
+        self.print_data = print_data
+        self.command = command
+
+        self.timestamps = dict[str,list]()
+        self.last_timestamp = dict[str]()
+        self.intervals = dict[str,list]()
+        self.values = dict[str,list]()
+        self.chunk_sizes = dict[str,list]()
+
+        for metric in metrics:
+            self.timestamps[metric] = list()
+            self.last_timestamp[metric] = None
+            self.intervals[metric] = list()
+            self.values[metric] = list()
+            self.chunk_sizes[metric] = list()
+
+
+        super().__init__(*args, client_version=client_version, **kwargs)
+
+    async def connect(self):
+        await super().connect()
+
+        click.echo(
+            click.style(
+                f"Inspecting the metric '{self._metrics}'...",
+                fg="green",
+            )
+        )
+
+        await self.subscribe(self._metrics)
+
+        await self.run_cmd()
+
+        await self.stop()
+        self.print_histograms()
+
+    async def run_cmd(self):
+        click.echo(f'running... {self.command!r}')
+
+        proc = await asyncio.create_subprocess_shell(
+            self.command,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE)
+
+        stdout, stderr = await proc.communicate()
+
+        click.echo(f'{self.command!r} exited with {proc.returncode}')
+        if stdout:
+            click.echo(stdout.decode())
+        if stderr:
+            click.echo(stderr.decode())
+
+    async def _on_data_message(self, message: aio_pika.IncomingMessage):
```

Comment:
Please add a comment why it is necessary to override `_on_data_message`.

I guess this is here to keep track of the sources that produce these metrics, right?

Reply:
wasn't necessary anymore and has been removed